src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -327,16 +327,7 @@ def setup(self, trainer: "pl.Trainer") -> None:
     def setup_optimizers(self, trainer: "pl.Trainer") -> None:
         if self.kwargs.get("use_orig_params"):
             return super().setup_optimizers(trainer)
-
-        invalid_params_error = False
-        try:
-            super().setup_optimizers(trainer)
-        except ValueError as ex:
-            if "optimizer got an empty parameter list" not in str(ex):
-                raise
-            invalid_params_error = True
```

Comment:
For `use_orig_params=False` FSDP configurations (which are certainly still supported), this change prevents `setup_optimizers` from ever being called.

I know the team is putting the finishing touches on the release but I think preventing FSDP from running in `use_orig_params=False` mode should be considered a release blocker and this should be refactored or reverted prior to release. 

Excited for all the new goodness coming with `2.1.0`, thanks for your continued superlative contributions to the open-source ML community!!
@carmocca @awaelchli @Borda

Reply:
Ah sorry @speediedan. Don't worry, we'll fix this before that

Reply:
awesome, thanks @carmocca !!!

Reply:
@carmocca I think it is what we were wondering yesterday. For PyTorch < 2.0, where use_orig_params does not exist, I think we need to to show this error so that the user can change their code to self.model.parameters() in the configure_optimizers() hook. And the way to check this in PyTorch < 2.0 was this try-except.

Reply:
Makes sense, we should add a test and put this under `if not use_orig_params`

Reply:
We could check `getattr(p, "._is_sharded", False)` (https://github.com/pytorch/pytorch/blob/v1.12.0/torch/distributed/fsdp/fully_sharded_data_parallel.py#L1329) which is what I chose to do in #18746